### PR TITLE
Pin versions for jaeger-hotrod docker-compose

### PR DESCRIPTION
**Pull request by fridgepoet**
### Description
There have been some updates in the images pulled for the Jaeger Hotrod demo's docker-compose which have broken the functionality, for example: https://github.com/opensearch-project/data-prepper/issues/2273#issuecomment-1827662450

I propose pinning versions which are working now and later explicitly changing the versions in the docker-compose when necessary.
 
### Issues Resolved
Related to https://github.com/opensearch-project/data-prepper/issues/2273
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: examples/jaeger-hotrod/docker-compose.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ services:
       - my_network
   opensearch:
     container_name: node-0.example.com
-    image: opensearchproject/opensearch:2
+    image: opensearchproject/opensearch:2.9.0
```
Comment (dlvenable) [inline examples/jaeger-hotrod/docker-compose.yml:57]: We should make the OpenSearch versions match. So update this to 2.11.0 or bring Dashboards to 2.9.0.

File: examples/jaeger-hotrod/docker-compose.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ services:
   data-prepper:
     restart: unless-stopped
     container_name: data-prepper
-    image: opensearchproject/data-prepper:2
+    image: opensearchproject/data-prepper:2.5.0
```
Comment (dlvenable) [inline examples/jaeger-hotrod/docker-compose.yml:6]: I tend to think that as this is a Data Prepper example, it should use the latest `2` version. We do try to ensure that we don't introduce breaking changes between versions. What do you think about keeping this one tied to `2`?

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thank you for this PR. I have also been thinking that we needed to pin those Jaeger versions as I've had some difficulty getting this to run as well.

File: examples/jaeger-hotrod/docker-compose.yml (hunk 1 above)
Comment (fridgepoet) [reply examples/jaeger-hotrod/docker-compose.yml:57]: Done! with 2.9.0

It seems like 2.11.0 is not working.

File: examples/jaeger-hotrod/docker-compose.yml (hunk 2 above)
Comment (fridgepoet) [reply examples/jaeger-hotrod/docker-compose.yml]: Makes sense, done!

**Review by dlvenable — APPROVED** [review]
Thank you @fridgepoet for this contribution.

**Review by kkondaka — APPROVED** [review]

**Pull request merged**